### PR TITLE
Testing Laravel

### PR DESCRIPTION
**Pull request by kawax**
We should set up a testing environment before working for Laravel.

- composer test / make test : Regular Tests. Except for Laravel.
- composer laravel-test / make laravel-test : Only Laravel. PHP>=7.2 Laravel>=6.0

Start with a small PR.

### CODE REVIEW

File: phpunit.xml [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
+    <filter>
+        <whitelist processUncoveredFilesFromWhitelist="true">
+            <directory suffix=".php">./src</directory>
+        </whitelist>
+    </filter>
     <php>
-        <env name="LINE_BOT_CHANNEL_ACCESS_TOKEN" value="test_channel_access_token"/>
-        <env name="LINE_BOT_CHANNEL_SECRET" value="test_channel_secret"/>
+        <server name="LINE_BOT_CHANNEL_ACCESS_TOKEN" value="test_channel_access_token"/>
+        <server name="LINE_BOT_CHANNEL_SECRET" value="test_channel_secret"/>
```
Comment (moririnson) [inline phpunit.xml:18]: Sorry for lack of my knowledges.
Can `env()` load value from `$_SERVER` too ?

Comment (moririnson) [reply phpunit.xml:18]: Are there no problems for use `server`?
We should test can load variables from `.env`.

Comment (kawax) [reply phpunit.xml]: I'm always referring to Laravel's code.
https://github.com/laravel/laravel/commit/69bd1dceefb1434e57dbbda7625c571030dd5edb

File: .travis.yml [hunk 2]
```diff
@@ -5,14 +5,23 @@ matrix:
...
+  - if [ "$TESTBENCH" != '' ]; then composer require "orchestra/testbench=$TESTBENCH" --dev --no-update; fi
+
 install:
-  - COMPOSER_MEMORY_LIMIT=-1 composer update
-script: make check
+  - COMPOSER_MEMORY_LIMIT=-1 composer install --no-progress --no-suggest --optimize-autoloader
+
+script:
+  - if [ "$TESTBENCH" = '' ]; then make check; fi
+  - if [ "$TESTBENCH" != '' ]; then make laravel-test; fi
```
Comment (moririnson) [inline .travis.yml:26]: don't need to exec `copyright phpcs phpmd`?

File: .travis.yml [hunk 3]
```diff
@@ -5,14 +5,23 @@ matrix:
...
-before_install: >
-  php_version=$(php --version);
-  if  [ ${php_version:4:3} = '5.5' ]; then
-    composer self-update --rollback
-  fi
+      env: TESTBENCH=^6.0
+
+before_install:
+  - if [ "$TESTBENCH" != '' ]; then composer remove phpunit/phpunit orchestra/testbench --dev --no-update; fi
+  - if [ "$TESTBENCH" != '' ]; then composer require "orchestra/testbench=$TESTBENCH" --dev --no-update; fi
```
Comment (moririnson) [inline .travis.yml:19]: can you change like this? （it looks like `if` statements duplicated.）

```
 >
  if [ "$TESTBENCH" = '' ];
    composer remove phpunit/phpunit orchestra/testbench --dev --no-update
    composer require "orchestra/testbench=$TESTBENCH" --dev --no-update
  fi
```

File: .travis.yml (hunk 2 above)
Comment (kawax) [reply .travis.yml:28]: make check includes src/Laravel

Comment (moririnson) [reply .travis.yml:28]: it looks `laravel-test` doesn't include `copyright phpcs phpmd`, right?
how to exec `copyright phpcs phpmd` when `"$TESTBENCH" = ''` ?

Comment (kawax) [reply .travis.yml:28]: I'll add laravel-check

**Review by Vaduz — COMMENTED** [review]
I couldn't find any improvements from your request.
These changes will make dependency more complex among the project (composer, Makefile, etc) and that's lessen readability.
In addition to that, Laravel specific tests are same as before.
If you want to use latest Laravel testing library, you should create another package for integrating with Laravel like `line-bot-sdk-php-laravel`. (We are not going to do it for now)

**Comment by kawax** [comment]
First, I suggested separating the SDK.
It's easy to make it as my personal library. But this needs to be released as an official SDK.
Unofficial libraries have a lot of mistakes. To improve it, need an official.

**Pull request closed without merging**